cpu/kinetis_common/periph/gpio.c

```diff
@@ -24,384 +24,257 @@
 
-/* Pool of linked list entries which can be used by any port configuration.
- * Rationale: Avoid dynamic memory inside low level periph drivers. */
-gpio_int_config_entry_t config_pool[GPIO_INT_POOL_SIZE];
+/**
+ * @brief   Cleaned up GPIO base address
+ */
+#define GPIO_ADDR_BASE      (GPIOA_BASE & ~(GPIO_ADDR_MASK))
 
-static PORT_Type * const _port_ptrs[] = PORT_BASE_PTRS;
-static GPIO_Type * const _gpio_ptrs[] = GPIO_BASE_PTRS;
+/**
+ * @brief   Kinetis CPUs have 32 pins per port
+ */
+#define PINS_PER_PORT       (32)
 
-static inline uint32_t _port_num(gpio_t dev) {
-    return (uint32_t)((dev & GPIO_PORT_MASK) >> GPIO_PORT_SHIFT);
-}
+/**
+ * @brief   Calculate the needed memory (in byte) needed to save 4 bits per MCU
+ *          pin
+ */
+#define ISR_MAP_SIZE        (GPIO_PORTS_NUMOF * PINS_PER_PORT * 4 / 8)
 
-static inline uint8_t _pin_num(gpio_t dev) {
-    return (uint8_t)((dev & GPIO_PIN_MASK) >> GPIO_PIN_SHIFT);
-}
+/**
+ * @brief   Define the number of simultaneously configurable interrupt channels
+ *
+ * We have configured 4-bits per pin, so we can go up to 16 simultaneous active
+ * extern interrupt sources.
+ */
+#define CTX_NUMOF           (8U)
 
-static inline PORT_Type *_port(gpio_t dev) {
-    return _port_ptrs[_port_num(dev)];
-}
+/**
+ * @brief   Interrupt context data
+ */
+typedef struct {
+    gpio_cb_t cb;
+    void *arg;
+    uint32_t state;
+} isr_ctx_t;
 
-static inline GPIO_Type *_gpio(gpio_t dev) {
-    return _gpio_ptrs[_port_num(dev)];
-}
+/**
+ * @brief   Allocation of memory for each independent interrupt slot
+ */
+static isr_ctx_t isr_ctx[CTX_NUMOF] = {{0}};
 
-static void _clear_interrupt_config(gpio_t dev) {
-    gpio_int_config_entry_t* entry = NULL;
-    uint8_t pin_number = _pin_num(dev);
-    mutex_lock(&int_config_lock);
-    /* Search for the given pin in the port's interrupt configuration */
-    LL_SEARCH_SCALAR(gpio_interrupts[_port_num(dev)], entry, pin, pin_number);
-    if (entry != NULL) {
-        LL_DELETE(gpio_interrupts[_port_num(dev)], entry);
-        /* pin == 0 means the entry is available */
-        entry->pin = 0;
-    }
-    mutex_unlock(&int_config_lock);
-}
+/**
+ * @brief   Allocation of 4 bit per pin to map a pin to an interrupt context
+ */
+static uint32_t isr_map[ISR_MAP_SIZE];
 
-static gpio_int_config_entry_t* _allocate_interrupt_config(uint8_t port) {
-    gpio_int_config_entry_t* ret = NULL;
-
-    mutex_lock(&int_config_lock);
-    for (uint8_t i = 0; i < GPIO_INT_POOL_SIZE; ++i) {
-        if (config_pool[i].pin == 0) {
-            /* temporarily set pin to something non-zero until the proper pin
-             * number is set by the init code */
-            config_pool[i].pin = 200;
-            ret = &config_pool[i];
-            LL_PREPEND(gpio_interrupts[port], ret);
-            break;
-        }
-    }
-    mutex_unlock(&int_config_lock);
 
-    return ret;
+static inline PORT_Type *port(gpio_t pin)
```

Comment:
Here and all other `static` functions: Could you add a prefix underscore to the name to follow the conventions used in other modules (e.g. gnrc, xtimer)